File: brownfield-backend/src/main/java/com/petclinic/invoice/Invoice.java
package com.petclinic.invoice;

import com.petclinic.visit.Visit;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import jakarta.persistence.JoinColumn;
import java.time.LocalDateTime;
import java.math.BigDecimal;

@Entity
public class Invoice {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String invoiceNumber;

    private LocalDateTime invoiceDate;

    private BigDecimal amount;

    @OneToOne
    @JoinColumn(name = "visit_id")
    private Visit visit;

    public Invoice() {
    }

    public Invoice(String invoiceNumber, LocalDateTime invoiceDate, BigDecimal amount, Visit visit) {
        this.invoiceNumber = invoiceNumber;
        this.invoiceDate = invoiceDate;
        this.amount = amount;
        this.visit = visit;
    }

    public Invoice(Long id, String invoiceNumber, LocalDateTime invoiceDate, BigDecimal amount, Visit visit) {
        this.id = id;
        this.invoiceNumber = invoiceNumber;
        this.invoiceDate = invoiceDate;
        this.amount = amount;
        this.visit = visit;
    }

    public Long getId() {
        return id;
    }

    public String getInvoiceNumber() {
        return invoiceNumber;
    }

    public LocalDateTime getInvoiceDate() {
        return invoiceDate;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public Visit getVisit() {
        return visit;
    }
}

File: brownfield-backend/src/invoice/invoice.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  OneToOne,
  JoinColumn,
} from "typeorm";
import { Visit } from "../visit/visit";

/**
 * @openapi
 * components:
 *   schemas:
 *     Invoice:
 *       type: object
 *       properties:
 *         id:
 *           type: integer
 *         invoiceNumber:
 *           type: string
 *         invoiceDate:
 *           type: string
 *           format: date-time
 *         amount:
 *           type: number
 *         visit:
 *           $ref: '#/components/schemas/Visit'
 */
@Entity()
export class Invoice {
  @PrimaryGeneratedColumn()
  readonly id!: number;

  @Column({ name: "invoice_number", type: "varchar" })
  readonly invoiceNumber!: string;

  @Column({ name: "invoice_date", type: "varchar" })
  readonly invoiceDate!: string;

  @Column({ type: "decimal", precision: 10, scale: 2 })
  readonly amount!: number;

  @OneToOne(() => Visit)
  @JoinColumn({ name: "visit_id" })
  readonly visit!: Visit;

  constructor(
    invoiceNumber?: string,
    invoiceDate?: string,
    amount?: number,
    visit?: Visit,
  ) {
    if (invoiceNumber !== undefined) this.invoiceNumber = invoiceNumber;
    if (invoiceDate !== undefined) this.invoiceDate = invoiceDate;
    if (amount !== undefined) this.amount = amount;
    if (visit !== undefined) this.visit = visit;
  }
}
